databuilder/justfile

```diff
@@ -0,0 +1,118 @@
+# make a new virtualenv so we don't use the main docs one, which is why we're
+# not using $VIRTUAL_ENV here.
+export VIRTUAL_ENV  := ".venv"
+
+# TODO: make it /scripts on windows?
```

Comment:
question: What about running any of this on Windows?

Again, not absolutely necessary now, but if we want non-developer contributors to add to this, then it might be something to think about.

An alternative might be to have a `workflow_dispatch` workflow that builds the snippets for you, (and maybe checks them first) and avoids a local checkout.

Reply:
The justfile stack doesn't support windows currently because of this line.  Simon and I have discussed it a few times and we think it should be simple to support with a condition on the OS, we've just not had the need yet (but assume it will come eventually).

Building examples with a workflow should be trivial, the workflow in this PR already does that, but it doesn't commit the change.  The only downside to consider with that is seeing the outcome of those changes in the docs would mean changes need to go through the workflow first.